Public/Src/Cache/ContentStore/Library/Utils/ResultsExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Diagnostics.ContractsLight;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Synchronization;
+
+namespace BuildXL.Cache.ContentStore.Utils
+{
+    internal static class ResultsExtensions
+    {
+        /// <nodoc />
+        public static TResult WithLockAcquisitionDuration<TResult, TLockKey>(this TResult result, in LockSet<TLockKey>.LockHandle handle)
+            where TResult : ResultBase
+            where TLockKey : IEquatable<TLockKey>
+        {
+            Contract.Requires(!result.Succeeded || result.Diagnostics == null, $"Diagnostics property can be set only once.");
+
+            if (result.Succeeded && handle.LockAcquisitionDuration != null)
+            {
+                var message = $", LockWait={(long)handle.LockAcquisitionDuration.Value.TotalMilliseconds}ms";

```

Comment:
>, [](start = 32, length = 1)

I don't think that this comma fits with any of our existing telemetry. Could this be removed?

Reply:
The comma actually perfectly fits into existing telemetry because this method is called from `PutResult.ToString()` method that uses the following approach: `$"Success Hash={ContentHash}, Size={ContentSize}"`. With this change it'll become: `Success Hash=hash, Size=42, LockWait=42ms`.

Reply:
The comma actually perfectly fits into existing telemetry because this method is called from PutResult.ToString() method that uses the following approach: $"Success Hash={ContentHash}, Size={ContentSize}". With this change it'll become: Success Hash=hash, Size=42, LockWait=42ms.

---
In reply to: [280619318](https://github.com/Microsoft/BuildXL/pull/233#discussion_r280619318) [](ancestors = 280619318)

Reply:
You're right, I was incorrect to suggest that it wouldn't fit with any of our existing telemetry. While it does fit with PutResult, it doesn't fit with OpenStreamResult and PlaceFileResult.

---
In reply to: [280638720](https://github.com/Microsoft/BuildXL/pull/233#discussion_r280638720) [](ancestors = 280638720,280619318)